pkg/controller/vitesskeyspace/reconcile_shards.go

```diff
@@ -187,6 +199,48 @@ func allShardsServingWrites(shardNames []string, shardStatus map[string]planetsc
 	return result
 }
 
+func totalReadytablets(shardNames []string, shardStatus map[string]planetscalev2.VitessKeyspaceShardStatus) int32 {
```

Comment:
My initial thought was that this looks copypasta-ish and all of this logic could sit in one subroutine or method? So you could call something like `updateTabletCounts()` to set all the values in one place.

Either way, this isn't that important to me and I won't block on it, just let me know what you feel about it.

Reply:
That's true to some extent but would require passing in the thing we want to mutate. I tend to avoid doing that because it can lead to buggy code that's unclear. I find it better usually to design pure functions that don't mutate data and return aggregated results. I agree it's less efficient, but I don't personally believe that two functions that do similar things are always bad.

Reply:
I'm totally cool with keeping it how it is, just wanted to throw that option out there just in case you had thought of it yourself and were on the edge of making the change or not :)